README.md

```diff
@@ -1,2 +1,217 @@
+   when (newState.type) {
+     StateType.Success -> success()
+     StateType.Error -> error()
+     StateType.Loading -> loading()
+   }
+}
+```
+
+
+### RxJob
+> `RxJob` is an implementation of `Job` that works with Reactive Extensions
+
+
+You just need to inherit your use case class from `RxJob` (specifying the type of the input parameter) and override the `bind(input: T):Completable` method with the properly work.
+
+>You must handle all the possible edge cases into your job. You should not allow your job to throw an exception back to the `Hub`.
+
+As it is returns a `Completable` you should handle the possible returns on the `RxStore` subscription, with `doOnSubscribe`, `doOnSuccess` and `doOnError` as shown in the next example:
+
+```kotlin
+class DoGoogleLoginJob @Inject constructor(
+    private val store: RxStore<LoginState>,
+    private val firebase: GoogleFirebaseAuth) : RxJob<Intent>() {
+
+  override fun bind(input: Intent): Completable {
+    return firebase.firebaseAuthWith(intent = input)
+        .doOnSubscribe { store.update { setType(StateType.Loading) } }
+        .doOnSuccess { store.update { setType(StateType.Success) } }
+        .doOnError { store.update { setType(StateType.Error) } }
+        .toCompletable()
+        .onErrorComplete()
+  }
+}
+```
+
+> In this example we are injecting the local parameters `store` and `firebase` with [dagger](https://github.com/google/dagger) we strong recommend that.
+
+### RxHub
+> `RxHub` is an implementation of `Hub` that works with Reactive Extensions
+
+You connect the `View` and the `RxJob`'s layers through the operator `bind(job: RxJob<T>)`(that's why all of your `View`'s methods needs to return an `Observable`).
+
+After that just need to inherit the class `RxHub<T>`(specifying the `View` it will connect with) and override the `connect(view: T)` method with the binds to `RxJobs` you need.
+
+We have already implemented the `disconnect()` method to release resources by disposing all the bound jobs.
+
+```kotlin
+class LoginHub @Inject constructor(
+    private val doGoogleLoginJob: RxJob<Intent>,
+    private val requestGoogleLoginJob: RxJob<Unit>) : RxHub<LoginView>() {
+
+  override fun connect(view: LoginView) {
+    view.doLoginClicks()
+        .bind(requestGoogleLoginJob)
+
+    view.activityResults()
+        .filter { it.first == GoogleLogin.GOOGLE_REQUEST }
+        .map { it.second }
+        .bind(doGoogleLoginJob)
+  }
+}
+```
+
+And finally that's how you connect and disconnect your `View` to the `Hub`
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_login)
+
+    ...
+    hub.connect(this)
+}
+
+override fun onDestroy() {
+    super.onDestroy()
+    hub.disconnect()
+}
+```
+
+---
+
+If you found potencial in this project please help us to improve the framework and help the open source community to have access to an awesome and well developed project!
```

Comment:
Suggestion:

We are looking forward to improving this framework. If you have some feedback, comments, questions or if you want to contribute, join us at our [Slack group](https://join.slack.com/t/fluent-io/shared_invite/enQtMzQ3MDY3Njk0MjI3LTZkMjU0Y2Q3ZTA2NWQ0OTE4ZjFiZmQxNzJlMTYxNjIwODBjYmJhOGRlYzRhNGY5NmNjZTVhMTIzMTJhNzEwYTY)